index.bs

```diff
@@ -11302,6 +11306,136 @@ The [=remote end steps=] given |session| and |command parameters| are:
+
+1. If installing |installation type| isn't supported return [=error=] with [=error code=] [=unsupported operation=].
+
+1. Switch on the value of the <code>installation type</code>:
+
+   <dl>
+    <dt>"<code>path</code>"
+    <dd>
+      1. Let |path| be a value of <code>path</code> field of |extension data spec|.
+      1. Let |entry| be a [=directory entry=] represeting the directory with that |path|.
+
+    <dt>"<code>archive-path</code>"
+    <dd>
+      1. Let |archive path| be a value of <code>path</code> field of |extension data spec|.
+      1. Perform any implementation-defined steps to extract ZIP from |archive path| and let |entry| by a [=directory entry=] representing the extraction result, if this fails return [=error=] with [=error code=] [=invalid web-extension=].
+
+    <dt>"<code>base64</code>"
+    <dd>
+      1. Let |archive path| be a value of <code>path</code> field of |extension data spec|.
+      1. Let |bytes| be [=forgiving-base64 decode=] |extension data spec|["<code>path</code>"].
+      1. Perform any implementation-defined steps to extract ZIP from |bytes| and let |entry| by a [=directory entry=] representing the extraction result, if this fails return [=error=] with [=error code=] [=invalid web-extension=].
+
+   </dl>
+
+1. Return [=success=] with data |entry|.
+
+</div>
+
+### Commands ### {#module-extensions-commands}
+
+#### The webExtensions.install Command #### {#command-extensions-install}
+
+The <dfn export for=commands>webExtensions.install</dfn> command installs an extension for the session.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      webExtensions.Install = (
+         method: "webExtensions.install",
+         params: webExtensions.InstallParameters
+      )
+
+      webExtensions.InstallParameters = {
+         extensionData: webExtensions.ExtensionPath / webExtensions.ExtensionArchivePath / webExtensions.ExtensionBase64Encoded,
+         ? allowPrivateBrowsing: bool .default false,
+      }
+
+      webExtensions.ExtensionPath = {
+         type: "path",
+         path: text,
+      }
+
+      webExtensions.ExtensionArchivePath = {
+         type: "archive-path",
+         path: text,
+      }
+
+      webExtensions.ExtensionBase64Encoded = {
+         type: "base64",
+         value: text,
+      }
+      </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+      <pre class="cddl local-cddl">
+      webExtensions.InstallResult = {
+        extension: text
+      }
+      </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for webExtensions.install">
+The [=remote end steps=] with |command parameters| are:
+
+1. If installing extensions isn't supported return [=error=] with error code [=unsupported operation=].
+
+1. Let |extension data spec| be a value of <code>extensionData</code> field of the |command parameters|.
+
+1. Let |extension directory entry| be the result of [=trying=] to [=expand a extension data spec=] with |extension data spec|.
```

Comment:
Should the `directory entry` part reference the other spec?

Reply:
isn't it enough if `|expand a extension data spec|` returns a `[=directory entry=]`?